benchmarks/benchmark_serving.py

```diff
@@ -577,38 +578,50 @@ def main(args: argparse.Namespace):
       )
   )
 
+  # Process output
+  output = [output.to_dict() for output in request_outputs]
+
   # Save config and results to json
   if args.save_result:
-    result_json = {}
+    # dimensions values are strings
+    dimensions_json = {}
+    # metrics values are numerical
+    metrics_json = {}
 
     # Setup
     current_dt = datetime.now().strftime("%Y%m%d-%H%M%S")
-    result_json["date"] = current_dt
-    result_json["model_id"] = model_id
-    result_json["tokenizer_id"] = tokenizer_id
-    result_json["num_prompts"] = args.num_prompts
+    dimensions_json["date"] = current_dt
+    dimensions_json["model_id"] = model_id
+    dimensions_json["tokenizer_id"] = tokenizer_id
+    metrics_json["num_prompts"] = args.num_prompts
 
     # Traffic
-    result_json["request_rate"] = (
+    metrics_json["request_rate"] = (
         args.request_rate if args.request_rate < float("inf") else "inf"
     )
 
-    # Merge with benchmark result
-    result_json = {**result_json, **benchmark_result}
+    metrics_json = {**metrics_json, **benchmark_result}
+    if args.run_eval:
+      eval_json = eval_accuracy(output)
+      metrics_json = {**metrics_json, **eval_json}
```

Comment:
It maybe work keeping metrics separated into "perf" and "eval" categories. What is the interface to XLML storage -- are you flattening all metrics into one table?

Reply:
The `metrics` json is a single-level json. During json-lines [processing](https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/xlml/utils/metric.py#L181), it uses the assumption that the key is a string and value is a float (not another json).

It populates BigQuery tables in normalized tables. I'll show you what I mean via chat.

Also, see example in this test https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/xlml/utils/metric_test.py#L155